main.tf

```diff
@@ -2,53 +2,44 @@ data "aws_caller_identity" "default" {}
 
 data "aws_region" "default" {}
 
-resource "aws_s3_bucket" "cache_bucket" {
+module "cache_bucket" {
   #bridgecrew:skip=BC_AWS_S3_13:Skipping `Enable S3 Bucket Logging` check until bridgecrew will support dynamic blocks (https://github.com/bridgecrewio/checkov/issues/776).
   #bridgecrew:skip=BC_AWS_S3_14:Skipping `Ensure all data stored in the S3 bucket is securely encrypted at rest` check until bridgecrew will support dynamic blocks (https://github.com/bridgecrewio/checkov/issues/776).
   #bridgecrew:skip=CKV_AWS_52:Skipping `Ensure S3 bucket has MFA delete enabled` due to issue in terraform (https://github.com/hashicorp/terraform-provider-aws/issues/629).
-  count         = module.this.enabled && local.create_s3_cache_bucket ? 1 : 0
-  bucket        = local.cache_bucket_name_normalised
-  acl           = "private"
-  force_destroy = true
-  tags          = module.this.tags
-
-  versioning {
-    enabled = var.versioning_enabled
+  source  = "cloudposse/s3-bucket/aws"
+  version = "2.0.3"
+
+  count       = module.this.enabled && local.create_s3_cache_bucket ? 1 : 0
+  bucket_name = local.cache_bucket_name_normalised
+
+  acl                = true
+  force_destroy      = true
+  tags               = module.this.tags
+  versioning_enabled = var.versioning_enabled
+  logging = {
+    bucket_name = var.access_log_bucket_name
+    prefix      = "logs/${module.this.id}/"
   }
-
-  dynamic "logging" {
-    for_each = var.access_log_bucket_name != "" ? [1] : []
-    content {
-      target_bucket = var.access_log_bucket_name
-      target_prefix = "logs/${module.this.id}/"
-    }
-  }
-
-  lifecycle_rule {
-    id      = "codebuildcache"
-    enabled = true
-
-    prefix = "/"
-    tags   = module.this.tags
-
-    expiration {
-      days = var.cache_expiration_days
-    }
-  }
-
-  dynamic "server_side_encryption_configuration" {
-    for_each = var.encryption_enabled ? ["true"] : []
-
-    content {
-      rule {
-        apply_server_side_encryption_by_default {
-          sse_algorithm = "AES256"
-        }
+  lifecycle_configuration_rules = [
+    # Be sure to cover https://github.com/cloudposse/terraform-aws-s3-bucket/issues/137
+    {
+      enabled                                = true
+      id                                     = "codebuildcache"
+      abort_incomplete_multipart_upload_days = 1
+      prefix                                 = "/"
```

Comment:
`prefix` is not a recognized element of `lifecycle_configuration_rules`. If needed, add to `filter_and` instead, but since prefix is `/` for the rule, no filter is needed.
```suggestion
```